### ISSUE TITLE
Clipping from continent osm.bpf returns empty osm.bpf

### ISSUE THREAD

**Issue by sahand-asgarpour** [osm-flex/osm-flex#25]
cp.clip_from_shapes is used with kernel='osmconvert' and overwrite=True for a given clipping shape.

**Comment by chahank** [comment]
Thank you for reaching out. Could you please post a minimal code to reproduce the issue? It is quite hard to diagnose otherwise.

**Comment by sahand-asgarpour** [comment]
Thank you for your quick response. Here you are:


root_folder = OSM_DATA_DIR.parent

region_code = "Europe"

dump_region = region_code.lower()

dump_folder = root_folder / "osm_bpf"

study_area_suffix = '_test'
clip_polygon_path = Path(
    rf'C:<some_path>\polygon{study_area_suffix}.geojson'
)
clip_output_name = f'study_area{study_area_suffix}'
study_area_dump_path = root_folder.joinpath('osm_bpf', f'{clip_output_name}.osm.pbf')

get_region_geofabrik(region=region_code, save_path=dump_folder)

with open(clip_polygon_path) as f:
    clip_polygon = geojson.load(f)

polygon_feature = clip_polygon['features'][0]
polygon_geom = shape(polygon_feature['geometry'])
cp.clip_from_shapes([polygon_geom],
                    osmpbf_output=OSM_DATA_DIR.joinpath(f'{clip_output_name}.osm.pbf'),
                    osmpbf_clip_from=OSM_DATA_DIR.joinpath(f'{dump_region}-latest.osm.pbf'),
                    kernel='osmconvert', overwrite=True)

**Comment by keesvanginkel** [comment]
It is maybe good to mention that this workflow did work for clipping country dumps (.osm.pbf), but somehow does not return anything in case the country dump is replaced with a Europe dump. Could it be something with the executable?

I remember from the past that we had problems **fetching** data from the Europe dump directly, but I thought that clipping normally worked.

**Comment by Evelyn-M** [comment]
Hi both
Without the error code output or data it's not possible to reproduce your issue, but for me clipping from a region file works.  Here's an example using only osm-flex methods and osmconvert as the clipping tool.
check if this works for you?
(The europe file is huge, so havent tried it yet, this example is on central america, clipping out the shape of honduras).

````
# Loading necessary data packages
import sys
sys.path.append('/Users/evelynm/osm-flex/src') # your path to osm-flex/src

import osm_flex.download as dl
import osm_flex.extract as ex
import osm_flex.config
import osm_flex.clip as cp

# Downloading a larger "parent" file to clip from: central america, from geofabrik.de 
# --> saved to OSM_DATA_DIR/central-america-latest.osm.pbf
dl.get_region_geofabrik('central-america')

# Obtain a country polygon (Switerland) and clip the europe file to it:
import shapely
admin_1_hnd = cp.get_admin1_shapes('HND')
admin_1_hnd = shapely.ops.unary_union([poly for poly in list(admin_1_hnd.values())]) # make a single valid polygon!
admin_1_hnd

cp.clip_from_shapes([admin_1_hnd], 
                    osmpbf_output=osm_flex.config.OSM_DATA_DIR.joinpath('hnd_clipped.osm.pbf'),
                    osmpbf_clip_from=osm_flex.config.OSM_DATA_DIR.joinpath('central-america-latest.osm.pbf'),
                    kernel='osmconvert', overwrite=True)
````

Two things to keep in mind: 
- is the polygon which should be clipped valid, not too complicated (.poly files, which are produced under the hood, simply end at one point if they are too long).
- is the polygon fully contained in the bigger region?

**Comment by sahand-asgarpour** [comment]
I ran the clipping in the command prompt and got this:

osmconvert Error: could not jump in file: C:\...\osm\osm_bpf\europe-latest.osm.pbf

I downloaded europe-latest.osm.pbf again and checked the permissions as well.

**Comment by Evelyn-M** [comment]
Hey Sahand.
It doesnt work because your polygon is located in central america, not in Europe (mind the negative longitudes - that is the "western" part of the world.)
It'll work with a polygon that is actually located in Europe (I checked ;) ).

Btw, for working with rectangular polygons, you can also use the method `cp.clip_from_bbox`

your bbox here would be: (-87.58411419237983,
 13.722375595429895,
 -86.69481007217836,
 14.319436681532025)
 
 Hope this helps, I'll close the issue later on.

**Comment by sahand-asgarpour** [comment]
I think the one you are refering to was used to clip the central america pbf.

For EU I used two polygons and both efforts failed with this error (when ran from the command prompt):
- osmconvert Error: could not jump in file: C:...\osm\osm_bpf\europe-latest.osm.pbf
Polygons:

{
	"type": "FeatureCollection",
	"features": [
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"coordinates": [
					[
						[
							4.3180613482170149,
							51.893671141359138
						],
						[
							4.296451513268551,
							51.89653460175441
						],
						[
							4.250970242551119,
							51.91003132119772
						],
						[
							4.231455593088754,
							51.922844797193139
						],
						[
							4.198092169587056,
							51.93885624869807
						],
						[
							4.131207947943551,
							51.97028029616507
						],
						[
							4.067994725922631,
							51.988426069030968
						],
						[
							3.962765745383251,
							52.0088086309552
						],
						[
							3.9509422644225369,
							51.922483230807447
						],
						[
							4.242982244123823,
							51.86374536989533
						],
						[
							4.334023047512744,
							51.86447550375823
						],
						[
							4.3180613482170149,
							51.893671141359138
						]
					]
				],
				"type": "Polygon"
			}
		}
	]
}


{
	"type": "FeatureCollection",
	"name": "POLYGON - Copy",
	"features": [
		{
			"type": "Feature",
			"properties": {},
			"geometry": {
				"type": "Polygon",
				"coordinates": [
					[
						[
							4.473665435221051,
							51.93383195597022
						],
						[
							4.431506973800339,
							51.923988605958928
						],
						[
							4.376660043406616,
							51.9330748518486
						],
						[
							4.333273665632163,
							51.92133810438139
						],
						[
							4.269836590340142,
							51.935845415216437
						],
						[
							3.979146346080742,
							52.01984522310225
						],
						[
							3.946055512671178,
							51.91700909466022
						],
						[
							4.180637344054617,
							51.889160221708838
						],
						[
							4.295103346792132,
							51.85004316738966
						],
						[
							4.564217735249072,
							51.85871831888491
						],
						[
							4.512516760794966,
							51.93046286550401
						],
						[
							4.473665435221051,
							51.93383195597022
						]
					]
				]
			}
		}
	]
}

**Comment by Evelyn-M** [comment]
Ah yes, true. `osmconvert` cannot handle such a large parent file. See the osmconvert documentation:

> Limitations: The Windows version (the downloaded binary) does not work with large files such as planet.osm (greater than 2Gb) when using the --complete-ways, --complete-multipolygons or --complete-boundaries option.

https://wiki.openstreetmap.org/wiki/Osmconvert

I think there's nothing one can do about it, except for switching the clipping kernel to `osmosis`.
This clips it for me, using your polygon above, in about 5 mins.

**Comment by Evelyn-M** [comment]
Reopening this, as there's a way around it. 
Actually, without the `--complete-` options in the command line command which it runs under the hood, `osmconvert` should also work with large parent files. 
@ElcoK knows how exactly, he can help out / fix the source code next week.

**Comment by Evelyn-M** [comment]
This took a while, but I removed the `--complete-ways` and `--complete-multipolygons` options now from the osmconvert command that are built ([abe21b7](https://github.com/osm-flex/osm-flex/pull/24/commits/abe21b71235d4d7d82271e52234dc4597dd0ed34)). The implications are explained in the clipping tutorial.
@sahand-asgarpour  I checked clipping a file from a large osm-pbf parent file (>>2 GB), and it seems to work. But happy to get feedback.
